deployment/config.go

```diff
@@ -44,11 +44,17 @@ type Config struct {
 	// Type of the DB instance.
 	DBInstanceType string `default:"db.r4.large" validate:"notempty"`
 	// Type of the DB instance - postgres or mysql.
-	DBInstanceEngine string `default:"aurora-postgresql" validate:"oneof:{aurora-mysql, aurora-postgresql}"`
+	DBInstanceEngine string `default:"aurora-postgresql" validate:"oneof:{aurora-mysql, aurora-postgresql, cockroachdb}"`
```

Comment:
The database settings are getting a bit complicated. I would suggest splitting the entire thing out to a separate struct like `ElasticSearchSettings`. In that, I would mainly want 2 sections:
1. Just the terraform related settings, which are needed to deploy a DB.
2. Everything else that will be needed, if terraform doesn't deploy a DB.

With that in mind, I think we can remove `cockroachdb` from the `DBInstanceEngine` and keep the values only to what Terraform can use. This will also have the advantage of removing cockroach version from variables.tf which wasn't really needed and causes more confusion as to what's really happening.

Then I would suggest to remove separate settings for host, port and dbname, and replace them with exactly what we have in our config.json - DriverName, DataSource, DataSourceSearchReplicas. This makes the entire thing unaware of any driver specific logic and just blindly pastes whatever is given to the system.